pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -105,84 +93,107 @@ def read_buffer(self, expected_length=0):
+        self.gate_time = value
+
+    gate_time = Instrument.control(
         ":ACQ:APER?",
         ":ACQ:APER %f",
-        "Gate time for one measurement in s.",
+        """Control gate time of one measurement in s (float strictly from 2e-9 to 1000).""",
         validator=strict_range,
-        values=[2e-9, MAX_MEASUREMENT_TIME],  # Programmer's guide 8-92
+        values=[MIN_GATE_TIME, MAX_GATE_TIME],  # Programmer's guide 8-92
     )
 
     format = Instrument.control(
         "FORM?",
         "FORM %s",
-        "Reponse format (ASCII or REAL).",
+        "Control response format ('ASCII' or 'REAL').",
         validator=strict_discrete_set,
         values=["ASCII", "REAL"],
     )
 
     interpolator_autocalibrated = Instrument.control(
         ":CAL:INT:AUTO?",
         "CAL:INT:AUTO %s",
-        "Controls if interpolators should be calibrated automatically.",
+        """Control if interpolators should be calibrated automatically (bool).""",
         strict_discrete_set,
         values={True: 1.0, False: 0.0},
         map_values=True,
     )
 
-    def configure_frequency_array_measurement(self, n_samples, channel):
+    def configure_frequency_array_measurement(self, n_samples, channel, back_to_back=True):
         """
         Configure the counter for an array of measurements.
 
         :param n_samples: The number of samples
         :param channel: Measurement channel (A, B, C, E, INTREF)
+        :param back_to_back: If True, the buffer measurement is performed back-to-back.
         """
-        n_samples = truncated_range(n_samples, [1, self.MAX_BUFFER_SIZE])
+        n_samples = truncated_range(n_samples, [MIN_BUFFER_SIZE, MAX_BUFFER_SIZE])
         channel = strict_discrete_set(channel, self.CHANNELS)
         channel = self.CHANNELS[channel]
-        self.write(f":CONF:ARR:FREQ {n_samples},(@{channel})")
+        self.write(f":CONF:ARR:FREQ{':BTB' if back_to_back else ''} {n_samples},(@{channel})")
 
     def buffer_frequency_time_series(
-        self, channel, n_samples, sample_rate, trigger_source=None
+        self,
+        channel,
+        n_samples,
+        sample_rate=None,  # deprecated, only kept for backwards compatibility
+        gate_time=None,
+        trigger_source=None,
+        back_to_back=True,
     ):
         """
         Record a time series to the buffer and read it out after completion.
 
         :param channel: Channel that should be used
         :param n_samples: The number of samples
+        :param gate_time: Gate time in s
+        :param trigger_source: Optionally specify a trigger source to start the measurement
+        :param back_to_back: If True, the buffer measurement is performed back-to-back.
         :param sample_rate: Sample rate in Hz
-        :param trigger_source: Optionally specify a trigger source to start the
-            measurement
+            .. deprecated:: 0.14
+                Use parameter `gate_time` instead.
```

Comment:
I think you have to add an empty line for sphinx (that is the linting error):

```suggestion

            .. deprecated:: 0.14
                Use parameter `gate_time` instead.
```

Reply:
Indeed! Passes the tests now.